cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/EvaluationOrderGraphPass.kt

```diff
@@ -191,15 +189,19 @@ open class EvaluationOrderGraphPass(ctx: TranslationContext) : TranslationUnitPa
             eogNodes
                 .filter { node ->
                     node is FunctionDeclaration ||
+                        node is VariableDeclaration ||
```

Comment:
Is this to not filter out eog s built for default values?

Reply:
> Yeah, this should prevent removing EOG edges for nodes which are not reachable by the EOG from the starting node for different reasons. Similar to the filters which already existed. I'm wondering though if we can use the `ResolutionStartHolder` interface instead of listing all these classes?

That might work. I'll try

Reply:
I changed it to 
```kotlin
var validStarts =
            eogNodes.filter { it is ResolutionStartHolder || it is VariableDeclaration }.toSet()
```

The tests in cog-core work, but I have no idea if this is "correct"